gwcs/wcstools.py

```diff
@@ -271,10 +273,83 @@ def wcs_from_points(xy, world_coordinates, fiducial,
         poly_x = fitter(poly, x, y, projection_x)
         poly_y = fitter(poly, x, y, projection_y)
     transform = models.Mapping((0, 1, 0, 1)) | poly_x & poly_y | projection.inverse | skyrot.inverse
-    
+
     skyframe = CelestialFrame(reference_frame=fiducial.frame)
     detector = Frame2D(name="detector")
     pipeline = [(detector, transform),
                 (skyframe, None)
                 ]
     return WCS(pipeline)
+
+
+def fit_wcs_from_points(xy, world_coordinates, proj_point=None, projection=None,
+                        degree=4, distortion_type='polynomial'):
+    """
+    Given two matching sets of coordinates on detector and sky, fit a WCS.
+
+    Notes
+    -----
+    This function implements the following algorithm:
+    ``world_coordinates`` are transformed to a projection plane using the specified projection.
+    A polynomial fits ``xy`` and the projected coordinates.
+    The fitted polynomials and the projection transforms are combined into a tranform
+    from detector to sky.
+    The input coordinate frame is set to ``detector``.
+    The output coordinate frame is initialized based on the frame in the fiducial.
+
+
+    Parameters
+    ----------
+    xy : tuple of 2 ndarrays
+        Points in the input cooridnate frame - x, y inputs.
+    world_coordinates : `~astropy.coordinates.SkyCoord` or tuple of 2 ndarrays
+        Points in the output coordinate frame.
+        The order matches the order of ``xy``.
+        If a tuple, it is assumed the numbers represent location in the same frame as `proj_point`.
+    proj_point : `~astropy.coordinates.SkyCoord` or str
+        A fiducial point in the output coordinate frame.
+    projection : `~astropy.modeling.projections.Projection`
+        A projection type. One of the projections in `~astropy.modeling.projections.projcode`.
+    degree : int
+        Degree of Polynpomial model to be fit to data.
+    distortion_type : str
```

Comment:
From the previous parameter `degree` it is clear that only polynomial type distortions are supported. Wouldn't it be more clear to rename this parameter to `polynomial_type` (as it was before) or `polynomial_form` (??) and I would suggest also renaming `"polynomial"` to `"standard"` (or something like that) in the next line.

Reply:
I would actually add an optional `None` value and allow for other type of distortions to be added later.

Reply:
@nden I would go another route: if there _ever_ will be another type of non-polynomial distortion (which I do not believe there will be), I would simply ignore `degree` when a non-polynomial distortion is requested. I think introduction of `None` just makes things more complicated for dubious _future_ benefits.

Reply:
Are you suggesting keeping a default `degree=4` ?

Reply:
yes. I think it would be better to keep it to 4. If you think you will ever have non-polynomial distortions, then keep `distortion_type` name (instead of suggested `polynomial_type`).